rfcs/react-components/styles-handbook.md

```diff
@@ -0,0 +1,771 @@
+- [Basics](#basics)
+- [APIs](#apis)
+  - [`makeStyles`](#makestyles)
+    - [Limitations](#limitations)
+  - [`mergeClasses()`](#mergeclasses)
+    - [Do not concatenate classes](#do-not-concatenate-classes)
+  - [`makeResetStyles`](#makeresetstyles)
+    - [Hybrid approach](#hybrid-approach)
+- [Advanced](#advanced)
+  - [Understanding selectors complexity](#understanding-selectors-complexity)
+- [Best practices](#best-practices)
+  - [Writting styles](#writting-styles)
+    - [Use `tokens` over direct colors](#use-tokens-over-direct-colors)
+    - [Avoid duplication in rule definitions](#avoid-duplication-in-rule-definitions)
+    - [Avoid `!important`](#avoid-important)
+  - [Performance](#performance)
+    - [Avoid nested `mergeClasses` calls](#avoid-nested-mergeclasses-calls)
+  - [Nested selectors](#nested-selectors)
+    - [Usage with pseudo classes](#usage-with-pseudo-classes)
+    - [Do not use selectors to target elements](#do-not-use-selectors-to-target-elements)
+    - [Do not introduce complicated selectors](#do-not-introduce-complicated-selectors)
+    - [Avoid usage of input pseudo classes](#avoid-usage-of-input-pseudo-classes)
+  - [RTL styles](#rtl-styles)
+    - [Using `@noflip`](#using-noflip)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# Introduction
+
+Griffel is a hybrid CSS-in-JS that features runtime option like any other CSS-in-JS and [Ahead-of-time compilation][griffel-aot] with [CSS extraction][griffel-css-extraction] to reduce runtime footprint and improve performance.
+
+# Basics
+
+Griffel uses Atomic CSS to generate classes. In Atomic CSS every property-value is written as a single CSS rule.
+
+```css
+/* Monolithic classes */
+/* Can be applied only to a specific button */
+.button {
+  display: flex;
+  align-items: center;
+}
+
+/* Atomic CSS */
+/* Can be applied to any element that needs there rules */
+.display-flex {
+  display: flex;
+}
+.align-items-center {
+  align-items: center;
+}
+```
+
+[Learn more][griffel-atomic-css] about Atomic CSS.
+
+# APIs
+
+## `makeStyles`
+
+Is used to define style permutations in components and is used for style overrides. Returns a [React hook][react-hook] that should be called inside a component:
+
+```jsx
+import { makeStyles } from '@griffel/react';
+
+const useClasses = makeStyles({
+  button: { display: 'flex' },
+  icon: { paddingLeft: '5px' },
+});
+```
+
+### Limitations
+
+Atomic CSS generates more classes than the approach with monolithic classes. Usually it's not a problem, but there are cases when performance [may degrade][griffel-recalc-performance], for example when elements have more than 100 classes. This does not happen usually as there is a limited set of CSS properties that is applied to a DOM element:
+
+```js
+import { makeStyles, shorthands } from '@griffel/react';
+
+const useClasses = makeStyles({
+  root: {
+    display: 'flex',
+    alignItems: 'center',
+    ...shorthands.padding('4px'),
```

Comment:
This is the first time `shorthands` is used.
1. This document expects the reader already understands how to use Griffel. That is fine, but I would just explicitly call it out in Basics chapter as a prerequisite.
2. Even after that, I would add one sentence about shorthands to the Basics as well.

Reply:
@miroslavstastny that's fair, I added it under limitations for `makeStyles` in fff7339.